pkg/infra/game/agent/agent.go

```diff
@@ -141,8 +148,14 @@ func (a *Agent) handleLootRoundMessage(
 ) {
 	switch r := m.Message().(type) {
 	case message.LootRequest:
-		req := *message.NewTaggedRequestMessage[message.LootRequest](m.Sender(), r, m.MID())
-		resp := a.Strategy.HandleLootRequest(req)
+		var resp message.Message
+		if a.MessagesRemaining < 1 {
+			resp = nil
```

Comment:
I think using nil here is a bad idea solely because we should tag the response with the sender such that the receiver of the response knows the agent can't respond

Reply:
Isn't this effectively the case anyway if they receive nil? 
It's unlikely (or maybe not-so-unlikely), albeit possible, that agents would reply to a request with nil. 
Maybe something more explicit like an 'Unavailable' message instead?

Reply:
Yes — but the sender is unknown when receiving a nil.